### PR TITLE
Partial #2237

### LINKED ISSUE #2237: Fix capitalization/wording in analysis plugin
All of the options in the dropdowns should follow standard capitalization and spacing rules, and b alphabetized. Adding small hints for now, until Qiime2 gets better, should also reduce user confusion.

Parameters in the plug in should be made user-friendly as they are in the Qiime1 processing plugin.

image
Should be:

Calculate alpha diversity (rarefied features only)
Calculate beta diversity (rarefied features only)
Filter samples by metadata
Rarefy features 
Summarize taxa (closed reference OTUs only)

image

Choose command [<- this is already correct]
Biom artifact:
Parameter set  [<- this is already correct]
Diversity metric
Phylogenetic tree

image

Abundance-based Coverage Estimator (ACE) metric
Berger-Parker dominance index
Brillouin's index
Chao 1 index
Chao 1 confidence interval
Dominance measure
Effective number of species (ENS)/Probability of intra-or interspecific encounter (PIE) metric 
Esty’s confidence interval
Fisher’s index
Gini index
Good’s coverage of counts
Heip’s evenness measure
Kempton-Taylor Q index 
Lladser's confidence interval
Lladser's point estimate
Margalef’s richness index
McIntosh's dominance index D
McIntosh’s evenness measure E
Menhinick’s richness index
Michaelis-Menten fit to rarefaction curve of observed OTUs
Number of distinct OTUs
Number of double occurrences (doubletons)
Number of single occurrences (singletons)
Number of observed OTUs, including singles and doubles
Pielou's evenness
Robbins’ estimator
Shannon's index
Simpson’s index
Simpson’s evenness measure E
Strong’s dominance index (Dw)

image

Choose command [<- this is already correct]
Biom artifact
Parameter set  [<- this is already correct]
Diversity metric
Phylogenetic tree

image

Bray–Curtis dissimilarity 
Canberra distance
Chebyshev distance
City-block distance
Correlation coefficient
Cosine similarity
Dice measure
Euclidean  distance
Hamming distance
Jaccard similarity index
Kulczynski dissimilarity index
Manhattan distance
Mahalanobis distance
Matching components
Rogers-Tanimoto distance
Russell-Rao coefficients
Soergel distance
Species-by-species Euclidean
Sokal-Michener coefficient
Sokal-Sneath index
Squared Euclidean distance
Weighted Minkowski metric
Yule index

image

Choose command [<- this is already correct]
Biom artifact
Parameter set  [<- this is already correct]

image

Choose command [<- this is already correct]
Biom artifact
Parameter set  [<- this is already correct]
Minimum feature frequency across samples
Minimum features per sample
SQLite WHERE-clause [provide link if possible]
Maximum feature frequency across samples
Maximum features per sample

image

Calculate alpha correlation

image
Choose command [<- this is already correct]
Biom artifact
Parameter set  [<- this is already correct]
Correlation method

image
Pearson
Spearman (Pearson with ranks)

image

Calculate correlation
Calculate group significance
Generate principal coordinates analysis (PCoA)

image

Choose command [<- this is already correct]
Distance Matrix artifact
Parameter set  [<- this is already correct]
Number of permutations
Correlation method
Metadata category 

image

Choose command [<- this is already correct]
Distance Matrix artifact
Parameter set  [<- this is already correct]
Number of permutations
Analysis method
     PERMANOVA
     ANOSIM
Metadata category
Comparison type
     Non-pairwise
     Pairwise

image

Custom-axis Emperor plot

image

Choose command [<- this is already correct]
PCoA artifact
Parameter set  [<- this is already correct]
Custom axis

### PR DESCRIPTION
**Pull request by josenavas**
Fixes #2237 partially. This makes sure that the different options in the dropdowns and the parameters name are shown sorted alphabetically.

### CODE REVIEW

**Comment by antgonza** [comment]
Could you add a screenshot of how they used to look and how they look now? Thanks!

**Comment by josenavas** [comment]
Before:
orderold
After:
ordernew

**Comment by antgonza** [comment]
code looks good but not sure if the sorting should be case sensitive ... @adswafford and @tkosciol do you guys have a preference?

**Comment by tkosciol** [comment]
I think it should not be case-sensitive. It's just a personal preference.

**Comment by josenavas** [comment]
Note that on the issue it also mentions to rename commands/options etc so they follow a more natural writing. For example, all commands will start with upper case so I think it doesn't really matter once everything is normalized :D

**Comment by antgonza** [comment]
I think the code shouldn't be case sensitive.

**Comment by josenavas** [comment]
Done:
screen shot 2017-10-20 at 1 46 28 pm

File: qiita_pet/templates/artifact_ajax/processing_artifact.html [hunk 1]
```diff
@@ -403,9 +410,11 @@
         var $colDiv = $('<div>').addClass('col-sm-3').appendTo($rowDiv);
         var sel = $('<select>').appendTo($colDiv).attr('id', 'command-sel').attr('name', 'command').addClass('form-control').attr('placeholder', 'Choose command...');
         sel.append($("<option>").attr('value', "").text("Choose command...").prop('disabled', true).prop('selected', true));
-        for(var i=0; i<data.commands.length; i++) {
-          if (data.commands[i].output.length != 0) {
-            sel.append($("<option>").attr('value', data.commands[i].id).text(data.commands[i].command));
+        var commands = data.commands;
+        commands.sort(function(a, b) {return a.command.localeCompare(b.command, 'en', {'sensitivity': 'base'});} );
+        for(var i=0; i<commands.length; i++) {
+          if (commands[i].output.length != 0) {
```
Comment (ElDeveloper) [inline qiita_pet/templates/artifact_ajax/processing_artifact.html]: Can you use `!==`

Comment (josenavas) [reply qiita_pet/templates/artifact_ajax/processing_artifact.html]: Done

**Comment by josenavas** [comment]
Thanks @ElDeveloper, the comparison functions (a vs b) are slightly different because they are operating over different objects, so they can't be generalized.

**Comment by ElDeveloper** [comment]
I see, ok, thanks!

**Pull request merged**